src/Decoder.c

```diff
@@ -4113,6 +4282,82 @@ static ZyanStatus ZydisNodeHandlerEvexB(const ZydisDecodedInstruction* instructi
     *index = instruction->raw.evex.b;
     return ZYAN_STATUS_SUCCESS;
 }
+
+static ZyanStatus ZydisNodeHandlerEvexND(ZydisDecoderContext* context, 
+    const ZydisDecodedInstruction* instruction, ZyanU16* index)
+{
+    ZYAN_ASSERT(context); // TODO: remove
+    ZYAN_UNUSED(context);
+    ZYAN_ASSERT(instruction);
+    ZYAN_ASSERT(index);
+
+    ZYAN_ASSERT(instruction->encoding == ZYDIS_INSTRUCTION_ENCODING_EVEX);
+    ZYAN_ASSERT(instruction->attributes & ZYDIS_ATTRIB_HAS_EVEX);
+
+    // TODO: Error conditions
+
+    *index = instruction->raw.evex.ND;
+    return ZYAN_STATUS_SUCCESS;
+}
+
+static ZyanStatus ZydisNodeHandlerEvexNF(ZydisDecoderContext* context, 
+    const ZydisDecodedInstruction* instruction, ZyanU16* index)
+{
+    ZYAN_ASSERT(context); // TODO: remove
+    ZYAN_UNUSED(context);
+    ZYAN_ASSERT(instruction);
+    ZYAN_ASSERT(index);
+
+    ZYAN_ASSERT(instruction->encoding == ZYDIS_INSTRUCTION_ENCODING_EVEX);
+    ZYAN_ASSERT(instruction->attributes & ZYDIS_ATTRIB_HAS_EVEX);
+
+    // Error conditions
+
+    if (instruction->raw.evex.z || context->vector_unified.L || 
+        (context->vector_unified.mask & 0x03))
+    {
+        return ZYDIS_STATUS_DECODING_ERROR;
+    }
+
+    // TODO: NF without ND filter -> evex.b = zero upper
+
+    *index = instruction->raw.evex.NF;
+    return ZYAN_STATUS_SUCCESS;
+}
+
+static ZyanStatus ZydisNodeHandlerEvexSCC(ZydisDecoderContext* context, 
+    ZydisDecodedInstruction* instruction, ZyanU16* index)
+{
+    ZYAN_ASSERT(context);
+    ZYAN_ASSERT(instruction);
+    ZYAN_ASSERT(index);
+
+    ZYAN_ASSERT(instruction->encoding == ZYDIS_INSTRUCTION_ENCODING_EVEX);
+    ZYAN_ASSERT(instruction->attributes & ZYDIS_ATTRIB_HAS_EVEX);
+
+    // Error conditions
+
+    if (instruction->raw.evex.z || context->vector_unified.LL)
+    {
+        return ZYDIS_STATUS_DECODING_ERROR;
+    }
+
+    // APX conditional CMP and TEST uses a special form of the EVEX prefix which reuses
+    // the `.V4` bit and the `.aaa` bits as the `.SCC` condition code and the `.vvvv` bits
+    // as the DFV specifier. Other than the regular `.vvvv` NDS/NDD specifier, the DFV
+    // specifier bits are not inverted.
+
+    // All APX conditional CMP and TEST instructions must have a SCC filter!
+
+    context->vector_unified.vvvv = (~context->vector_unified.vvvv) & 0x0F;
```

Comment:
This is clumsy.. It would be nice if the generator emits a hint about the EVEX variant to use. We are currently just "guessing".